### PR TITLE
[RPC] Transaction RPC#229

### PR DESCRIPTION
**Pull request by wow-sven**
1.getTransactionByHash(vec[hash])
2.getTransactionByIndex(start, limit)

TODO:Optimize the executeTransaction Response

### CODE REVIEW

File: crates/rooch-sequencer/src/actor/transaction_list.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            current = node.next.as_ref();
+        }
+        None
+    }
+
+    pub async fn get_transaction_by_index(
+        &self,
+        start: u64,
+        limit: u64,
+    ) -> Option<Vec<TypedTransaction>> {
```
Comment (jolestar) [inline crates/rooch-sequencer/src/actor/transaction_list.rs:44]: Option<Vec<TypedTransaction>> -> `Vec<TypedTransaction>`, if no result, just return empty vec.

File: crates/rooch/src/commands/transaction/commands/get_by_hash.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+use crate::types::{CommandAction, WalletContextOptions};
+use async_trait::async_trait;
+use hex::FromHex;
+use rooch_server::jsonrpc_types::TransactionView;
+use rooch_types::{error::RoochResult, H256};
+
+#[derive(Debug, clap::Parser)]
+pub struct GetByHashCommand {
+    #[clap(long)]
+    pub hash: String,
```
Comment (jolestar) [inline crates/rooch/src/commands/transaction/commands/get_by_hash.rs:13]: use `H256`, `H256` implemented `FromStr`, and support the `0x` prefix or not.

File: crates/rooch-sequencer/src/actor/transaction_list.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    pub next: Option<Box<MemTransaction>>,
+}
+
+pub struct MemTransactionList {
+    pub head: Option<Box<MemTransaction>>,
+}
+
+impl MemTransactionList {
+    pub fn add_transaction(&mut self, transaction: TypedTransaction) {
+        let mut new_node = Box::new(MemTransaction {
```
Comment (baichuan3) [inline crates/rooch-sequencer/src/actor/transaction_list.rs:19]: If use a linked list for mock storage, it means that even if use hash as a key to query, it needs to traverse the linked list

File: crates/rooch-server/src/jsonrpc_types/rooch_types.rs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct TransactionView {
+    pub transaction_type: TransactionType,
+    // Sequence number of this transaction corresponding to sender's account.
+    pub sequence_number: u64,
+    pub sender: String,
+    // The MoveAction to execute.
+    pub action: MoveAction,
```
Comment (jolestar) [inline crates/rooch-server/src/jsonrpc_types/rooch_types.rs:16]: MoveAction also needs a MoveActionView.

File: crates/rooch-server/src/jsonrpc_types/rooch_types.rs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+use moveos_types::transaction::MoveAction;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct TransactionView {
+    pub transaction_type: TransactionType,
+    // Sequence number of this transaction corresponding to sender's account.
+    pub sequence_number: u64,
+    pub sender: String,
```
Comment (jolestar) [inline crates/rooch-server/src/jsonrpc_types/rooch_types.rs:14]: TBD: how to represent the sender. Add a TODO

File: crates/rooch-server/src/jsonrpc_types/rooch_types.rs [hunk 6]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use rooch_types::transaction::{TransactionType, TypedTransaction};
+
+use moveos_types::transaction::MoveAction;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct TransactionView {
```
Comment (jolestar) [inline crates/rooch-server/src/jsonrpc_types/rooch_types.rs:10]: We may need to put the original transaction bytes in the View. Convenient for the client to parse the Transaction.

File: crates/rooch-sequencer/src/actor/transaction_list.rs (hunk 3 above)
Comment (jolestar) [reply crates/rooch-sequencer/src/actor/transaction_list.rs:19]: We should define a transaction storage trait and implement memory storage. The memory storage is just a mock storage, we don't need to consider memory capacity and performance issues.

**Comment by jolestar** [comment]
Need write a test case in the https://github.com/rooch-network/rooch/blob/5408355c1ff77f8c9c9c8c4904452668c6ceafeb/crates/testsuite/features/cmd.feature

**Review by jolestar — APPROVED** [review]

**Pull request merged**